presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -673,7 +674,73 @@ public Node visitShowColumns(SqlBaseParser.ShowColumnsContext context)
     @Override
     public Node visitShowStats(SqlBaseParser.ShowStatsContext context)
     {
-        return new ShowStats(getLocation(context), getQualifiedName(context.qualifiedName()));
+        return new ShowStats(Optional.of(getLocation(context)), getQualifiedName(context.qualifiedName()));
+    }
+
+    @Override
+    public Node visitShowStatsParametrized(SqlBaseParser.ShowStatsParametrizedContext context)
+    {
+        // The following properties of SELECT subquery are required:
```

Comment:
> These should not be parser checks. They should happen during analysis of the parsed tree.

`SHOW` queries does not make to till analysis phase right now. We mimicked current approach which I agree is not the best one. 

> Also, why aren't these supported, and why is the child not just a query?

The intention was to support only filter as the mechanism was to show the statistics of the table scan, with possible pushed down predicate. I think it should be extended to support arbitrary query and return computed statistics estimates if we pass some complex query as an input. Followup?